tools/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/templates/SuperClassNameResolver.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.ui.templates;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.template.java.CompilationUnitContext;
+import org.eclipse.jface.text.templates.TemplateContext;
+import org.eclipse.jface.text.templates.TemplateVariableResolver;
+
+/**
+ * @author Joye Luo
+ */
+@SuppressWarnings( "restriction" )
+public class SuperClassNameResolver extends TemplateVariableResolver
+{
+
+    public SuperClassNameResolver()
+    {
+        super( "super_class_name", "get superclass name" );
+    }
+
+    @Override
+    protected String resolve( TemplateContext context )
+    {
+        String superClassName = "";
+
+        ICompilationUnit unit = ( (CompilationUnitContext) context ).getCompilationUnit();
+
+        String typeName = JavaCore.removeJavaLikeExtension( unit.getElementName() );
+
+        IType type = unit.getType( typeName );
+
+        try
+        {
+            superClassName = type.getSuperclassName();
+        }
+        catch( JavaModelException e )
+        {
+        }
+
+        return superClassName;
+    }
+
+    @Override
+    protected boolean isUnambiguous( TemplateContext context )
```

Comment:
1) for the unit.getType(), it will get main type by type name, and I want to get the public class name, so I used 
   typeName = JavaCore.removeJavaLikeExtension( unit.getElementName() );
 to get the public class name, so maybe I should change the typename into publicClassName.
2) for the function isUnambiguous(TemplateContext context ), is used to not show the  edit box when use this variable in a template.